include/dfu/dfu_target_smp.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file dfu_target_smp.h
+ *
+ * @defgroup dfu_target_SMP SMP external MCU target
+ * @{
+ * @brief DFU Target for upgrades performed by SMP Client
+ */
+
+#ifndef DFU_TARGET_SMP_H__
+#define DFU_TARGET_SMP_H__
+
+#include <stddef.h>
+#include <dfu/dfu_target.h>
+#include <zephyr/mgmt/mcumgr/grp/img_mgmt/img_mgmt_client.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief DFU target reset callback for activating MCUboot serial recovery mode.
+ *
+ * @return 0 on success, negative errno otherwise.
+ */
+typedef int (*dfu_target_reset_cb_t)(void);
+
+/**
+ * @brief Register recovery mode reset callback
+ *
+ * Function that boots the target in MCUboot serial recovery mode, needed before sending
+ * SMP commands.
+ *
+ * @param[in] cb User defined target reset function for entering recovery mode
+ *
+ * @return 0 on success, negative errno otherwise.
+ */
+int dfu_target_smp_recovery_mode_enable(dfu_target_reset_cb_t cb);
+
+/**
+ * @brief Initialize dfu target SMP client
+ *
+ * @retval 0 If successful, negative errno otherwise.
+ */
+int dfu_target_smp_client_init(void);
+
+/**
+ * @brief Read Image list
+ *
+ * @param res_buf Image List buffer
+ *
+ * @return 0 on success, negative errno otherwise.
+ */
+int dfu_target_smp_image_list_get(struct mcumgr_image_state *res_buf);
+
+/**
+ * @brief Reboot SMP target device and apply new image.
+ *
+ * @return 0 on success, negative errno otherwise.
+ */
+int dfu_target_smp_reboot(void);
+
+/**
+ * @brief Confirm activated new Image after reset command
+ *
+ * Use only without MCUBoot recovery mode
+ *
+ * @return 0 on success, negative errno otherwise.
+ *
+ */
+int dfu_target_smp_confirm_image(void);
+
+/**
+ * @brief See if data in buf indicates MCUBoot style upgrade.
```

Comment:
```suggestion
 * @brief Check if data in buffer indicates MCUboot style upgrade.
```
Do you mean **buffer** or, e.g., **buf parameter**?